modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -1298,6 +1319,93 @@ private CompletableFuture<Boolean> onPrimaryExpired(PrimaryReplicaEventParameter
             return falseCompletedFuture();
         }
 
+        // TODO: move to Replica https://issues.apache.org/jira/browse/IGNITE-23750
+        private void registerFailoverCallback(TablePartitionId replicationGroupId) {
+            CompletableFuture<Replica> replicaFuture = replicaManager.replica(replicationGroupId);
+
+            // TODO: https://issues.apache.org/jira/browse/IGNITE-23753
+            // For now we assume that it's a possible situation to get null future on node's stopp process, but it should be fixed with
+            // a correct and consistent replica's state.
+            if (replicaFuture == null) {
+                return;
+            }
+
+            replicaFuture.thenApply(electedPrimaryReplica -> {
```

Comment:
Here and there busy locks are missing.

Reply:
Hm, `ReplicaStateManager` don't have one. And it doesn't have (yet) `shutdown` method for turning busy lock on. There are options:
1. Do we really need it?
2. We may reuse `ReplicaManager`'s one?
3. Or add the special one with `shutdown()` that will call on `ReplicaManager#stopAsync`?

Reply:
1. I guess yes.
2. I guess yes.

Up to you.

Reply:
Did with local `ReplicaStateManager#busyLock`, but I wish that I do it right